File: python/6 kyu/332.py
# https://www.codewars.com/kata/55031bba8cba40ada90011c4/train/python

# Passed

def is_sum_of_cubes(s):
    is_cubic = lambda num_group: sum(int(digit) ** 3 for digit in num_group) == int(num_group)
    current_group = str()
    number_groups = []
    for char in s:
        if char.isdigit() and len(current_group) < 3:
            current_group += char
        elif current_group:
            number_groups.append(current_group)
            current_group = char if char.isdigit() else str()
    
    if current_group:
        number_groups.append(current_group)
    
    cubic_groups = [number_group for number_group in number_groups if is_cubic(number_group)]
    result = "Unlucky"
    if len(cubic_groups) > 0:
        cubic_groups_sum = sum(int(cubic_group) for cubic_group in cubic_groups)
        result = f"{' '.join(cubic_groups)} {cubic_groups_sum} Lucky"
        
    return result

output = is_sum_of_cubes("&z _upon 407298a --- ???ry, ww/100 I thought, 631str*ng and w===y -721&()")
print(output)

File: nodejs/7 kyu/332.js
// https://www.codewars.com/kata/54d472e98776e4eb5b000215/train/javascript

// Passed

function pitchClass(note) {
  let noteToPitchClass = {
    C: 0,
    D: 2,
    E: 4,
    F: 5,
    G: 7,
    A: 9,
    B: 11,
  };
  let functionToApply = (val) => val;
  if (note.includes("#")) {
    functionToApply = (val) => (val + 1 > 11 ? 0 : val + 1);
  } else if (note.includes("b")) {
    functionToApply = (val) => (val - 1 < 0 ? 11 : val - 1);
  }

  note = note.replace("#", "").replace("b", "");
  let result =
    noteToPitchClass[note] !== undefined
      ? functionToApply(noteToPitchClass[note])
      : null;
  return result;
}

const output = pitchClass("D#");
console.log(output);